clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -59,4 +50,47 @@ public void iterator() throws Exception {
         }
         assertEquals(2, c);
     }
+
+    @Test
+    public void testRecordsWithNullTopic() {
+        String nullTopic = null;
+        Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new LinkedHashMap<>();
+        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> consumerRecords.records(nullTopic));
+        assertEquals("Topic must be non-null.", exception.getMessage());
+    }
+
+    @Test
+    public void testRecords() {
+        String targetTopic = "topic";
+        Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = buildTestRecordsWithDummyRecords(targetTopic);
```

Comment:
Could you build the `ConsumerRecords` with multiple topics? Also, we should verify the record for each topic.

Reply:
Sure, will do it 😸